linera-execution/src/execution.rs

```diff
@@ -232,10 +245,12 @@ where
         };
         // Set the authenticated signer to be used in outgoing messages.
         result.authenticated_signer = signer;
-        *remaining_fuel = runtime.remaining_fuel();
-        WASM_FUEL_USED_PER_BLOCK
-            .with_label_values(&[])
-            .observe((initial_remaining_fuel - *remaining_fuel) as f64);
+        let runtime_counts = runtime.runtime_counts();
+        let balance = self.system.balance.get_mut();
+        update_limits(balance, runtime_limits, pricing, runtime_counts)?;
```

Comment:
Should this be a method `update` of `RuntimeLimits`?

Reply:
Yes.